### ISSUE TITLE
ImagePullWorker pulls non-stop

### ISSUE THREAD

**Issue by matti** [kontena/kontena#1731]
With 1.1.0.rc1

kontena-lb deployed with stack:
```
stack: matti/lb-ingress
version: 0.1.0
expose: lb
variables:
  lb_stats_password:
    type: string
    from:
      vault: LB_INGRESS_STATS_PASSWORD
      random_string: 12
    to:
      vault: LB_INGRESS_STATS_PASSWORD
services:
  lb:
    image: kontena/lb:latest
    ports:
      - 80:80
      - 443:443
    affinity:
      - label==lb-ingress
    deploy:
      strategy: daemon
      wait_for_port: 80
      interval: 1h
    health_check:
      protocol: http
      port: 80
      uri: /__health
    environment:
      - KONTENA_LB_HEALTH_URI=/__health
    secrets:
      - secret: LB_INGRESS_STATS_PASSWORD
        name: STATS_PASSWORD
        type: env
```

Causes (?) this (see timestamps):

```
$ kontena grid logs -t | grep ImagePullWorker
kontena-agent | I, [2017-01-29T20:45:50.932935 #1]  INFO -- Kontena::Workers::ImagePullWorker: pulling image: kontena/lb:latest
kontena-agent | I, [2017-01-29T20:45:52.066893 #1]  INFO -- Kontena::Workers::ImagePullWorker: pulled image: kontena/lb:latest
kontena-agent | I, [2017-01-29T20:46:04.737800 #1]  INFO -- Kontena::Workers::ImagePullWorker: pulling image: kontena/lb:latest
kontena-agent | I, [2017-01-29T20:46:05.797409 #1]  INFO -- Kontena::Workers::ImagePullWorker: pulled image: kontena/lb:latest
kontena-agent | I, [2017-01-29T20:46:06.431687 #1]  INFO -- Kontena::Workers::ImagePullWorker: pulling image: kontena/lb:latest
kontena-agent | I, [2017-01-29T20:46:07.497297 #1]  INFO -- Kontena::Workers::ImagePullWorker: pulled image: kontena/lb:latest
kontena-agent | I, [2017-01-29T20:46:24.923399 #1]  INFO -- Kontena::Workers::ImagePullWorker: pulling image: kontena/lb:latest
kontena-agent | I, [2017-01-29T20:46:26.165511 #1]  INFO -- Kontena::Workers::ImagePullWorker: pulled image: kontena/lb:latest
kontena-agent | I, [2017-01-29T20:46:26.625723 #1]  INFO -- Kontena::Workers::ImagePullWorker: pulling image: kontena/lb:latest
kontena-agent | I, [2017-01-29T20:46:27.744637 #1]  INFO -- Kontena::Workers::ImagePullWorker: pulled image: kontena/lb:latest
```

**Comment by matti** [comment]
still pulls

**Comment by SpComb** [comment]
![2017-01-31-234433_962x113_scrot](https://cloud.githubusercontent.com/assets/3950373/22485688/44941096-e80f-11e6-8ac4-3ccefbf4fd54.png)

**Comment by SpComb** [comment]
I'm guessing the master thinks the service isn't running and is trying to re-deploy it in a loop, what does the `kontena service ls` show as the service status? Any related deployment messages in the server logs?

Alternatively some bug inside the Agent, but I'm not seeing that.

**Comment by matti** [comment]
Nope. running 2/1 (as daemon) as expected. Nothing else on the logs.

**Comment by matti** [comment]
I have two completely separate grids (one in azure, one in google cloud) that both have this problem - there might be some clues in master logs, have to dig deeper some day.

**Comment by matti** [comment]
This might have something to do with my experimental setup where master is also node?

**Comment by SpComb** [comment]
This is probably caused by https://github.com/kontena/kontena/issues/1895

**Comment by SpComb** [comment]
Does the agent image-pulling happen every 20s with associated `ServiceBalancerJob: rebalancing service: ..` messages in the master `kontena-server` logs?

**Comment by matti** [comment]
@SpComb yes.

```
master_1   | I, [2017-02-27T06:06:52.687606 #6]  INFO -- ServiceBalancerJob: rebalancing service: mygrid/lb-ingress/lb
master_1   | I, [2017-02-27T06:07:12.368641 #6]  INFO -- ServiceBalancerJob: rebalancing service: mygrid/lb-ingress/lb
master_1   | I, [2017-02-27T06:07:32.044737 #6]  INFO -- ServiceBalancerJob: rebalancing service: mygrid/lb-ingress/lb
master_1   | I, [2017-02-27T06:07:51.729709 #6]  INFO -- ServiceBalancerJob: rebalancing service: mygrid/lb-ingress/lb
master_1   | I, [2017-02-27T06:08:11.408344 #6]  INFO -- ServiceBalancerJob: rebalancing service: mygrid/lb-ingress/lb
master_1   | I, [2017-02-27T06:08:31.170813 #6]  INFO -- ServiceBalancerJob: rebalancing service: mygrid/lb-ingress/lb
master_1   | I, [2017-02-27T06:08:50.850771 #6]  INFO -- ServiceBalancerJob: rebalancing service: mygrid/lb-ingress/lb
master_1   | I, [2017-02-27T06:09:10.534194 #6]  INFO -- ServiceBalancerJob: rebalancing service: mygrid/lb-ingress/lb
```

**Comment by jakolehm** [comment]
#1873 should fix this.

**Comment by SpComb** [comment]
Fixed by #1873, the agent no longer pulls the image when checking if the container is up to date.
